File: c#/program11.cs
using System;

class NumberX
{
    public int No;

    public NumberX(int no)
    {
        this.No = no;
    }

    public int SumFactor()
    {
        int sum = 0;

        for(int iCnt = 1; iCnt <= (this.No/2); iCnt++)
        {
            if((this.No%iCnt) == 0)
            {
                sum = sum + iCnt;
            }

            if(sum > this.No)
            {
                break;
            }
        }
        return sum;
    }

    public bool CheckPerfect()
    {
        int result = 0;

        result = SumFactor();
        if(result == this.No)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
}

class program11
{
    public static void Main()
    {
        int value = 0;
        bool bRet = false;

        Console.WriteLine("Enter number : ");
        value = int.Parse(Console.ReadLine());

        NumberX obj = new NumberX(value);
        bRet = obj.CheckPerfect();

        if(bRet == true)
        {
            Console.WriteLine("Number is Perfect..");
        }
        else
        {
            Console.WriteLine("Number is NOT Perfect..");
        }
    }
}

File: scala/program11.scala
import scala.io.StdIn._

class Number
{
    var num : Int = 0

    def this(num : Int) {
        this()
        this.num = num;
    }

    def sumFactors(): Int = {
        var sum: Int = 0

        for(i <- 1 to (this.num/2))
        {
            if(num % i == 0) {
                sum += i;
            }
        }
        sum;
    }

    def checkPerfect(): Boolean = {
        var result = sumFactors();
        var flag = if(result == this.num) true else false
        flag;
    }
}

object Demo extends App
{
    println("Enter number to check its perfect or not : ")
    var num  = readInt()

    var obj = new Number(num)
    if (obj.checkPerfect()) {
        println("Number is perfect..")
    } else {
        println("Not perfect...")
    }
}